### PR TITLE
[INLONG-10208][Sort] ClsSink support unified configuration

### LINKED ISSUE #10208: [Feature][Sort] ClsSink support unified configuration
### Description

_No response_

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by vernedeng**
Fixes #10208 

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/v2/SinkContext.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    protected final Context sinkContext;
+
+    protected SortTaskConfig sortTaskConfig;
+
+    protected final Channel channel;
+    //
+    protected final int maxThreads;
+    protected final long processInterval;
+    protected final long reloadInterval;
+    //
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/v2/SinkContext.java:63]: Remove it.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/v2/SinkContext.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+    protected final String clusterId;
+    protected final String taskName;
+    protected final String sinkName;
+    protected final Context sinkContext;
+
+    protected SortTaskConfig sortTaskConfig;
+
+    protected final Channel channel;
+    //
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/v2/SinkContext.java:59]: Remove it.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+
+            // init
+            client.init();
+            // temporary use to ACK fetched msg.
+            callback.setClient(client);
+            return client;
+        } catch (UnknownHostException ex) {
+            LOG.error("got one UnknownHostException when init client of id:{}", sortTaskName, ex);
+        } catch (Throwable th) {
+            LOG.error("got one throwable when init client of id:{}", sortTaskName, th);
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:201]: Remove it.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                LOG.info("create sort sdk client in file way:{}", configType);
+                ClassResourceQueryConsumeConfig queryConfig = new ClassResourceQueryConsumeConfig();
+                client = SortClientFactory.createSortClient(clientConfig, queryConfig);
+            } else if (SortClusterConfigType.MANAGER.name().equalsIgnoreCase(configType)) {
+                LOG.info("create sort sdk client in manager way:{}", configType);
+                clientConfig.setManagerApiUrl(ManagerUrlHandler.getSortSourceConfigUrl());
+                client = SortClientFactory.createSortClient(clientConfig);
+            } else {
+                LOG.info("create sort sdk client in custom way:{}", configType);
+                // user-defined
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:178]: What is the purpose of this annotation?

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+
+    private String taskName;
+
+    // Context of SortSdkSource.
+    private SortSdkSourceContext context;
+
+    // The cluster name of sort.
+    private String sortClusterName;
+
+    // Reload config interval.
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:102]: Remove it.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java [hunk 6]
```diff
@@ -0,0 +1,224 @@
...
+
+    // Default consume strategy of {@link SortClient}.
+    private static final SortClientConfig.ConsumeStrategy defaultStrategy = SortClientConfig.ConsumeStrategy.lastest;
+
+    private static final String KEY_SORT_SDK_CLIENT_NUM = "sortSdkClientNum";
+    private static final int DEFAULT_SORT_SDK_CLIENT_NUM = 1;
+
+    private String taskName;
+
+    // Context of SortSdkSource.
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:96]: Remove it.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java [hunk 7]
```diff
@@ -0,0 +1,224 @@
...
+    private SortSdkSourceContext context;
+
+    // The cluster name of sort.
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:99]: Remove it.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java [hunk 8]
```diff
@@ -0,0 +1,224 @@
...
+    private String sortClusterName;
+
+    // Reload config interval.
+    private long reloadInterval;
+
+    // Executor for config reloading.
```
Comment (fuweng11) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:105]: Remove it.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/v2/SinkContext.java (hunk 1 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/v2/SinkContext.java:63]: fixed

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java (hunk 3 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:201]: fixed

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java (hunk 4 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:178]: fixed

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java (hunk 7 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:99]: fixed

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java (hunk 8 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:105]: fixed

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/v2/SinkContext.java [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+import org.apache.commons.lang3.StringUtils;
+import org.apache.flume.Channel;
+import org.apache.flume.Context;
+import org.slf4j.Logger;
+
+import java.util.Date;
+import java.util.Map;
+import java.util.Timer;
+import java.util.TimerTask;
+
```
[the remark this answers is not in the document]
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/v2/SinkContext.java:39]: Not all fields need to be set or get

**Review by fuweng11 — APPROVED** [review]

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSink.java [hunk 10]
```diff
@@ -57,13 +47,10 @@ public void start() {
                 worker.start();
             }
         } catch (Exception e) {
-            LOG.error(e.getMessage(), e);
+            LOG.error("failed to start cls sink", e);
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSink.java:50]: It's better "LOG.error("failed to start cls sink,error:{}", e.getMessage(), e);".
Log4j will skip exception message in repeat exception.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java [hunk 11]
```diff
@@ -104,26 +97,25 @@ public void reload() {
...
             });
 
-            SortTaskConfig newSortTaskConfig = SortClusterConfigHolder.getTaskConfig(taskName);
+            SortTaskConfig newSortTaskConfig = SortConfigHolder.getTaskConfig(taskName);
             if (newSortTaskConfig == null || newSortTaskConfig.equals(sortTaskConfig)) {
                 return;
             }
             LOG.info("get new SortTaskConfig:taskName:{}:config:{}", taskName,
                     new ObjectMapper().writeValueAsString(newSortTaskConfig));
             this.sortTaskConfig = newSortTaskConfig;
```
Comment (luchunliang) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java:106]: If requestNodeConfig.getVersion() < clsNodeConfig.getVersion(), reload function can directly return.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSink.java (hunk 10 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSink.java:50]: It's an optimization of JVM, uses **-XX:-OmitStackTraceInFastThrow** to forbid this option

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java (hunk 11 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java:106]: Both NodeConfig and DataflowConfig have their version. Must check all versions of dataflow and node.

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java [hunk 12]
```diff
@@ -0,0 +1,202 @@
...
+    private ScheduledExecutorService pool;
+
+    private List<SortClient> sortClients = new ArrayList<>();
+
+    @Override
+    public synchronized void start() {
+        int sortSdkClientNum = CommonPropertiesHolder.getInteger(KEY_SORT_SDK_CLIENT_NUM, DEFAULT_SORT_SDK_CLIENT_NUM);
+        LOG.info("start to SortSdkSource:{}, client num is {}", taskName, sortSdkClientNum);
+        for (int i = 0; i < sortSdkClientNum; i++) {
+            this.sortClients.add(this.newClient(taskName));
```
Comment (aloyszhang) [inline inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/source/sortsdk/v2/SortSdkSource.java:98]: `newClient()` may return null, so NPE may happened when travase the `sortClients`

File: inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java (hunk 11 above)
Comment (vernedeng) [reply inlong-sort-standalone/sort-standalone-source/src/main/java/org/apache/inlong/sort/standalone/sink/cls/ClsSinkContext.java:106]: The config update problem will be fixed in #10224

**Review by aloyszhang — APPROVED** [review]

**Review by luchunliang — APPROVED** [review]

**Review by fuweng11 — APPROVED** [review]

**Pull request merged**